### ISSUE TITLE
Extract core functionality into an NPM Module

### ISSUE THREAD

**Issue by sgoggins** [backyourstack/backyourstack#468]
Hi, 

I am new to this project and run it locally. I see a lot of great functionality for scanning repositories for some widely used languages for dependencies using requirements.txt (Python), package.json (Node stack/javascript), etc. 

I am wondering if this could be refactored into an NPM installable module that projects like one I work on, Augur, could use.  ( https://github.com/chaoss/augur ) ... I am impressed with what's possible, and think that kind of refactoring would make it easier for people who work on Augur to make upstream contributions back to this project. 

Sean

**Comment by RichardLitt** [comment]
This sounds like a good idea, to me. @znarf How much work do you think that would be?

**Comment by coderberry** [comment]
I believe the extraction wouldn't be terribly difficult. The code is very clean and I believe the majority of what we need is in the `src/scripts` folder

**Comment by znarf** [comment]
It depends what's the exact goal of the library and how its API looks like. What would be the key features of such library and what would be the data input? And data output?

Some pseudo-code of its usage would be welcome to see what you have in mind!

**Comment by RichardLitt** [comment]
> It depends what's the exact goal of the library and how its API looks like. 

Makes sense to me. I think just the functionality in scripts/* that check for deps, but don't touch collective data. I believe the input would be the name of a repo or an org, with a token, output would be a list of deps. @sgoggins might have more ideas here.
